devices/lidl.js

```diff
@@ -529,12 +711,152 @@ module.exports = [
+                .withFeature(exposes.binary('friday', ea.STATE_SET, 'ON', 'OFF'))
+                .withFeature(exposes.binary('saturday', ea.STATE_SET, 'ON', 'OFF'))
+                .withFeature(exposes.binary('sunday', ea.STATE_SET, 'ON', 'OFF')),
+            ...[1, 2, 3, 4, 5, 6].map((timeSlotNumber) =>
+                exposes
+                    .composite(`schedule_slot_${timeSlotNumber}`, `schedule_slot_${timeSlotNumber}`, ea.STATE_SET)
+                    .withDescription(`Watering time slot ${timeSlotNumber}`)
+                    .withFeature(
+                        exposes.binary('state', ea.STATE_SET, 'ON', 'OFF').withDescription('On/off state of the time slot'),
+                    )
+                    .withFeature(
+                        exposes
+                            .numeric('start_hour', ea.STATE_SET)
+                            .withUnit('h')
+                            .withValueMin(0)
+                            .withValueMax(23)
+                            .withDescription('Starting time (hour)'),
+                    )
+                    .withFeature(
+                        exposes
+                            .numeric('start_minute', ea.STATE_SET)
+                            .withUnit('min')
+                            .withValueMin(0)
+                            .withValueMax(59)
+                            .withDescription('Starting time (minute)'),
+                    )
+                    .withFeature(
+                        exposes
+                            .numeric('timer', ea.STATE_SET)
+                            .withUnit('min')
+                            .withValueMin(1)
+                            .withValueMax(599)
+                            .withDescription('Auto off after specific time for scheduled watering.'),
+                    )
+                    .withFeature(
+                        exposes
+                            .numeric('pause', ea.STATE_SET)
+                            .withUnit('min')
+                            .withValueMin(0)
+                            .withValueMax(599)
+                            .withDescription('Pause after each iteration.'),
+                    )
+                    .withFeature(
+                        exposes
+                            .numeric('iterations', ea.STATE_SET)
+                            .withValueMin(1)
+                            .withValueMax(9)
+                            .withDescription('Number of watering iterations. Works only if there is a pause.'),
+                    ),
+            ),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, null, valueConverterLocal.wateringState],
+                // disable optimistic state reporting (device may not turn on when battery is low)
+                [1, 'state', tuya.valueConverter.onOff, {optimistic: false}],
+                [5, 'timer', tuya.valueConverter.range(1, 599)],
```

Comment:
The range is already specified in the exposes, therefore there is no need to do it again here. I suggest to use `tuya.valueConverter.raw` here and remove `tuya.valueConverter.range`

Reply:
Yeah, I used this before but I noticed there isn't any validation. At least I was able to set unsupported values via frontend and MQTT messages. But that's a different story.